PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppRunState.java

```diff
@@ -180,6 +214,18 @@ private WebApp getWebAppAccordingToConfiguration(@NotNull RunProcessHandler proc
         return webApp;
     }
 
+    private int uploadWebConfigFile(@NotNull FTPClient ftp, @NotNull WebApp webApp, @NotNull String fileType,
+                                    @NotNull RunProcessHandler processHandler) throws IOException {
+        if (Comparing.equal(fileType, MavenConstants.TYPE_JAR) && webApp.operatingSystem() == OperatingSystem.WINDOWS) {
```

Comment:
Revert to previous method, but update the condition to always upload web.config for Windows web app if deploying jar artifact.